### PR TITLE
Define round for complex numbers

### PR DESCRIPTION
**Pull request by jiahao**

### CODE REVIEW

**Comment by jiahao** [comment]
@simonbyrne's more recent work on exposing `RoundingMode`s has lent more weight to the idea that only `round` should be exposed.

The C99 standard actually specifies separate rounding modes for the real and imaginary components, which seems a little odd at first, but I think is the only reasonable thing to do. `trunc`, `ceil` and `floor` correspond to three different real rounding modes which can be freely applied to real and imaginary components separately; in principle one could define `trunc(z)` in terms of the `(RoundToZero, RoundToZero)` rounding mode, but this seems to have limited value.

**Comment by jiahao** [comment]
Following what we do now for real numbers, I've exposed 3 different `round` methods for complex numbers:

- The "expert" rounding mode aware `round(z, RoundingMode, RoundingMode)`
  which takes separate RoundingModes for real and imaginary components
- The basic `round(z)` which applies round to the real and imaginary
  components separately using the current global rounding mode
- `round(z, d, b)` for rounding the real and imaginary components
  separately to d digits each in base b.

The last method, together with `float(z)`, is sufficient to restore the ability to round complex matrices with `round`.

**Comment by simonbyrne** [comment]
This seems like a reasonable approach.

File: doc/stdlib/math.rst [hunk 1]
```diff
@@ -666,6 +666,18 @@ Mathematical Functions
 
    ``round(x, digits)`` rounds to the specified number of digits after the decimal place, or before if negative, e.g., ``round(pi,2)`` is ``3.14``. ``round(x, digits, base)`` rounds using a different base, defaulting to 10, e.g., ``round(pi, 1, 8)`` is ``3.125``.
 
+.. function:: round(x, RoundingMode)
+
+   Returns the nearest integral value of the same type as ``x`` to ``x``,
+   breaking ties using the specified :obj:`RoundingMode`.
+
+.. function:: round(z, RoundingMode1, RoundingMode2)
```
Comment (ivarne) [inline doc/stdlib/math.rst]: I think the signature should rather be 

`round(z, RoundingModeReal, RoundingModeImaginary)`

Comment (ArchRobison) [inline doc/stdlib/math.rst]: +1

**Pull request merged**